lib/osf-components/addon/components/file-browser/delete-file-modal/template.hbs

```diff
@@ -0,0 +1,103 @@
+<OsfDialog as |dialog|>
+    <dialog.trigger>
+        <Button
+            @type='destroy'
+            {{on 'click' dialog.open}}
+        >
+            <FaIcon @icon='trash'/>
+            {{t 'general.delete'}}
+        </Button>
+    </dialog.trigger>
+    <dialog.heading>
+        {{#if this.shouldShowFailureModal}}
+            {{t 'osf-components.file-browser.delete_file_modal.delete_failed' itemCount=this.failedFiles.length}}
+        {{else if this.shouldShowSuccessModal}}
+            {{t 'osf-components.file-browser.delete_file_modal.delete_success' itemCount=this.deletedFiles.length}}
+        {{else}}
+            {{t 'osf-components.file-browser.delete_file_modal.heading' itemCount=@items.length}}
+        {{/if}}
+    </dialog.heading>
+    <dialog.main>
+        <ul local-class='ItemList'>
+            {{#if this.shouldShowFailureModal}}
+                {{#each this.failedFiles as |item|}}
+                    <li local-class='ListItem'>
+                        <div>
+                            {{#if item.isFile}}
+                                <FaIcon @icon='file-alt' />
+                            {{else}}
+                                <FaIcon @icon='folder' />
+                            {{/if}}
+                            <span local-class='ItemName'>{{item.name}}</span>
+                        </div>
+                        <Button
+                            @layout='fake-link'
+                            {{on 'click' (perform this.retryDelete item)}}
+                        >
```

Comment:
This should not be an active button if the retryDelete task is running. It should probably be a pulsed spinner with no clickability.